backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/BPACredentialDefinition.java

```diff
@@ -0,0 +1,46 @@
+package org.hyperledger.bpa.model;
+
+import io.micronaut.data.annotation.AutoPopulated;
+import io.micronaut.data.annotation.DateCreated;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.time.Instant;
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@Table(name = "bpa_cred_def")
+public class BPACredentialDefinition {
+
+    public final static Integer REVOCATION_REGISTRY_MIN_SIZE = 4;
+    public final static Integer REVOCATION_REGISTRY_MAX_SIZE = 32768;
+
+    @Id
+    @AutoPopulated
+    private UUID id;
+
+    @DateCreated
+    private Instant createdAt;
+
+    @ManyToOne
+    private BPASchema schema;
+
+    private String credentialDefinitionId;
+
+    private String tag;
+
+    private Boolean isSupportRevocation = Boolean.FALSE;
+
+    private Integer revocationRegistrySize = REVOCATION_REGISTRY_MIN_SIZE;
```

Comment:
I'm not sure if this works if you use the @Builder so you might end up writing null to the DB, and while reading the row the result is set to min size which is totally confusing and might end in unexpected behaviour. It's better to explicitly set this in the business logic of the manager.

Reply:
Seems no easy way to have NoArgs and Builder working properly without duplication. Tried overriding the Builder with default values, but that triggers pod violations. We do expect that the only data for this comes from the API and default values will not be used.

Reply:
Revocation registry size is a tricky param in any case. Too small and you will have lots of files, too big and you will end up having huge files and registry creation takes ages. So I would even propose not exposing that one at all. In another project that only cares about issuing we have set this to a fixed value of 3000. So in any case I would remove it from here, and then do any of the following:

1. also remove from api and statically set it to 3000
2. leave it in the api, but fallback to 3000 if not set
3. enforce a value to be set in the api, and never set it in the backend at all

Reply:
Interesting.  OK, I think hiding that makes sense, and less to explain to an end user! I think if we make it a configurable  default value, that gives enough flexibility per deployment.

Reply:
Removed from API, is now a runtime configuration for the default size.

Reply:
ok, can you remove the min and max fields as well as they are not used any more

Reply:
done.